Changed token enum to a Token Class

98adamdo opened the pull request:
Created a token class. Added a new string parameter "name" to include the name of the Token, as Riley wanted the token text stored in it. (I could be interpreting him wrong, and the parameter could be removed as it seems redundant). 
Created a build method as workaround to the "new" keyword being unable to be used in a nested Class. 
Instead of Token BOOLEAN = new Token(0), it's Token BOOLEAN = build(0, "BOOLEAN");

98adamdo committed:
Changed token enum to a Token Class

Created a token class. Added a new string parameter "name" to include the name of the Token, as Riley wanted the token text stored in it. (I could be interpreting him wrong, and the parameter could be removed as it seems redundant). 
Created a build method as workaround to the "new" keyword being unable to be used in a nested Class. 
Instead of Token BOOLEAN = new Token(0), it's Token BOOLEAN = build(0, "BOOLEAN");

RKuttruff commented:
Close. The String isn't the type of the token (though it'd still be wise to have that for a given token type) the string is the value (if any) of the token.

Example: (token - type - String)
true - _boolean - "true"
17 - _integer - "17"
if - _keyword - "if"
retVal - _identifier - "retVal"
( - _openParen - ""

Hope this makes sense.

I think you use the method yytext() or something like that to get the matched text for a token.

RKuttruff committed:
Update toy.flex

Fixed Token class, cleaned up regex's and rules, removed standalone, added variables for completion and errors (with getters), added error rule, misc. fixes

RKuttruff committed:
Update toy.flex

Fixed token type strings

RKuttruff committed:
Update toy.flex

Edited whitespace formatting

RKuttruff committed:
Update toy.flex

null is more appropriate for valuless tokens than empty string

RKuttruff committed:
Update toy.flex

Added error token for printing error messages
